### DISCUSSION TITLE
How to handle floating windows' dialogs. They get hidden under main window.

### DISCUSSION THREAD

**Discussion by benibilme** [i3/i3#5436] [Q&A]
Hello,

I am using i3, sometimes I need full blown file manager and for that I use dolphin.  I have following configuration in my i3 configuration file. I use dolphin as strachpad as follows.

````
for_window [class="dolphin"]  floating enable, move position center, sticky enable, move scratchpad
bindsym $mod+f  [instance="dolphin"] scratchpad show 
````

Dolphin behaves as expected. However all its dialogs are get hidden under main window. I can  not see them until I close the main window with the key combination. Then the dialog is accessible. Dolphins dialogs have the same window class as "dolphin" and I guess they all are scratchpad.  

Any help much appreciated.

**Comment by okraits** [comment]
I think you are using some redundant commands here. A scratchpad window is always floating and centered by default, so you can omit those two commands.
Why should the dolphin window be sticky? This is kind of against the usecase of the scratchpad (quickly show and hide a window).

**Comment by orestisfl** [comment]
> Dolphins dialogs have the same window class as "dolphin" and I guess they all are scratchpad.

Then it's probable that you can avoid matching them

Can you paste the output of `xprop` for the main window and for a dialog window? You can probably use the `window_type` criterion from https://i3wm.org/docs/userguide.html#command_criteria
